eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -55,19 +55,27 @@ parameters:
   - name: DisplayName
     type: string
     default: Validate
-
+  - name: OSName
+    type: string
+    default: ''
 
 jobs:
-  - job:
+  - job: ${{ parameters.DisplayName }}_${{ parameters.OSName }}
     displayName: ${{ parameters.DisplayName }}
     dependsOn: ${{ parameters.DependsOn }}
     condition: and(succeededOrFailed(), ne(variables['Skip.Test'], 'true'), ne(${{ parameters.Matrix }}, '{}'))
     strategy:
       matrix: $[ ${{ parameters.Matrix }} ]
       maxParallel: 12
     pool:
-      vmImage: $(OSVmImage)
       name: $(Pool)
+      # 1es pipeline templates converts `image` to demands: ImageOverride under the hood
+      # which is incompatible with image selection in the default non-1es hosted pools
+      ${{ if eq(parameters.OSName, 'macOS') }}:
```

Comment:
Should this condition be based on the pool name?

Reply:
Can't do compile time parameters using variables. We have the `OS` hardcoded at compile time, which is why we use it.